src/devices/bus/qbus/dvk_mx.cpp

```diff
@@ -0,0 +1,748 @@
+
+	LOGLIVE("%s: Called live_start (%d)\n", cur_live.tm.to_string(), state);
+	live_run();
+}
+
+void dvk_mx_device::checkpoint()
+{
+	if (cur_live.fi)
+		cur_live.pll.commit(cur_live.fi->dev, cur_live.tm);
+	checkpoint_live = cur_live;
+}
+
+void dvk_mx_device::rollback()
+{
+	cur_live = checkpoint_live;
+}
+
+void dvk_mx_device::live_delay(int state)
+{
+	cur_live.next_state = state;
+	if (cur_live.tm != machine().time())
+		cur_live.fi->tm->adjust(cur_live.tm - machine().time(), cur_live.fi->id);
+	else
+		live_sync();
+}
+
+void dvk_mx_device::live_sync()
+{
+	if (!cur_live.tm.is_never())
+	{
+		if (cur_live.tm > machine().time())
+		{
+			rollback();
+			live_run(machine().time());
+			cur_live.pll.commit(cur_live.fi->dev, cur_live.tm);
+		}
+		else
+		{
+			cur_live.pll.commit(cur_live.fi->dev, cur_live.tm);
+			if (cur_live.next_state != -1)
+			{
+				cur_live.state = cur_live.next_state;
+				cur_live.next_state = -1;
+			}
+			if (cur_live.state == IDLE)
+			{
+				cur_live.pll.stop_writing(cur_live.fi->dev, cur_live.tm);
+				cur_live.tm = attotime::never;
+				cur_live.fi->live = false;
+				cur_live.fi = nullptr;
+			}
+		}
+		cur_live.next_state = -1;
+		checkpoint();
+	}
+}
+
+void dvk_mx_device::live_abort()
+{
+	if (!cur_live.tm.is_never() && cur_live.tm > machine().time())
+	{
+		rollback();
+		live_run(machine().time());
+	}
+
+	if (cur_live.fi)
+	{
+		cur_live.pll.stop_writing(cur_live.fi->dev, cur_live.tm);
+		cur_live.fi->live = false;
+		cur_live.fi = nullptr;
+	}
+
+	cur_live.tm = attotime::never;
+	cur_live.state = IDLE;
+	cur_live.next_state = -1;
+}
+
+void dvk_mx_device::live_run(attotime limit)
+{
+	if (cur_live.state == IDLE || cur_live.next_state != -1)
+		return;
+
+	if (limit == attotime::never)
+	{
+		if (cur_live.fi->dev)
+			limit = cur_live.fi->dev->time_next_index();
+		if (limit == attotime::never)
+		{
+			// Happens when there's no disk or if the fdc is not
+			// connected to a drive, hence no index pulse. Force a
+			// sync from time to time in that case, so that the main
+			// cpu timeout isn't too painful.  Avoids looping into
+			// infinity looking for data too.
+
+			limit = machine().time() + attotime::from_msec(1);
+			cur_live.fi->tm->adjust(attotime::from_msec(1), cur_live.fi->id);
+		}
+	}
+
+	for (;;)
+	{
+		switch (cur_live.state)
+		{
+		case SEARCH_ID:
+			if (read_one_bit(limit))
+				return;
+
+			if (!(cur_live.bit_counter & 255))
+			LOGSHIFT("%s (%s): shift = %04x data=%02x c=%d\n", cur_live.tm.to_string(), limit.to_string(), cur_live.shift_reg,
+					(cur_live.shift_reg & 0x4000 ? 0x80 : 0x00) |
+					(cur_live.shift_reg & 0x1000 ? 0x40 : 0x00) |
+					(cur_live.shift_reg & 0x0400 ? 0x20 : 0x00) |
+					(cur_live.shift_reg & 0x0100 ? 0x10 : 0x00) |
+					(cur_live.shift_reg & 0x0040 ? 0x08 : 0x00) |
+					(cur_live.shift_reg & 0x0010 ? 0x04 : 0x00) |
+					(cur_live.shift_reg & 0x0004 ? 0x02 : 0x00) |
+					(cur_live.shift_reg & 0x0001 ? 0x01 : 0x00),
+					cur_live.bit_counter);
```

Comment:
Please use braces when the controlled statement occupies multiple lines.  Also, please indent the controlled statement.

You can simplify the bit extraction as `bitswap<8>(cur_live.shift_reg, 14, 12, 10, 8, 6, 4, 2, 0)`.